### PR TITLE
[[ Bug 22879 ]] Use correct text color for active tab button on Big Sur 

### PR DESCRIPTION
**Pull request by livecodeian**
This patch updates the MacOS implementation of MCPlatformGetControlThemePropColor to change the color returned for the text of a selected tab button when running on Big Sur, which now uses the standard control text color.

Fixes https://quality.livecode.com/show_bug.cgi?id=22879

### CODE REVIEW

**Review by runrevmark — CHANGES_REQUESTED** [review]
The OS version stuff is much neater 👍 

Perhaps make the tag on the commit title for the osversion commit `[[ OSVersion ]]` though (and also in the private repo).

**Comment by runrevmark** [comment]
@livecodepanos Could you take a look through the OSVersion commit here too (approve on github if it looks fine) - just make sure the hex value used is the same as the replaced code :)

**Review by runrevmark — APPROVED** [review]

File: engine/src/dskw32.cpp [hunk 1]
```diff
@@ -365,7 +365,7 @@ bool MCS_registry_root_to_hkey(MCStringRef p_root, HKEY& r_hkey, uint32_t& x_acc
 		if (MCStringIsEqualToCString(p_root, Regkeys[i].token, kMCCompareCaseless))
 		{
 			r_hkey = Regkeys[i].key;
-			if (MCmajorosversion >= 0x0501)
+			if (MCmajorosversion >= MCOSVersionMake(5,1,0))
```
Comment (livecodepanos) [inline engine/src/dskw32.cpp:368]: I think this should be `if (MCmajorosversion >= MCOSVersionMake(5,0,1))`

Comment (runrevmark) [reply engine/src/dskw32.cpp:368]: I think its correct - windows only actually encoded major.minor before in the hex values - although @livecodeian will need to confirm that

File: engine/src/w32date.cpp [hunk 2]
```diff
@@ -384,7 +384,7 @@ static void windows_cache_locale(void)
 	s_datetime_locale -> date_formats[2] = windows_query_date_format(LOCALE_SLONGDATE, false);
 
 	// AL-2013-02-08: [[ Bug 9942 ]] Allow appropriate versions of Windows to retrieve the short time format.
-	if (MCmajorosversion >= 0x0601)
+	if (MCmajorosversion >= MCOSVersionMake(6,1,0))
```
Comment (livecodepanos) [inline engine/src/w32date.cpp:387]: I think this should be `if (MCmajorosversion >= MCOSVersionMake(6,0,1))`

File: engine/src/w32cursor.cpp [hunk 3]
```diff
@@ -165,7 +165,7 @@ void MCScreenDC::resetcursors(void)
 	MCdispatcher -> clearcursors();
 
 	// XP and above allow alpha-blended cursors
-	if (MCmajorosversion >= 0x0501)
+	if (MCmajorosversion >= MCOSVersionMake(5,1,0))
 	{
```
Comment (livecodepanos) [inline engine/src/w32cursor.cpp:169]: I think this should be `if (MCmajorosversion >= MCOSVersionMake(5,0,1))`

File: engine/src/srvwindows.cpp [hunk 4]
```diff
@@ -292,7 +292,7 @@ struct MCWindowsSystem: public MCSystemInterface
 
 	virtual bool GetVersion(MCStringRef& r_string)
 	{
-		return MCStringFormat(r_string, "NT %d.%d", (MCmajorosversion >> 8) & 0xFF, MCmajorosversion & 0xFF);
+		return MCStringFormat(r_string, "NT %d.%d", MCOSVersionGetMajor(MCmajorosversion), MCOSVersionGetMinor(MCmajorosversion));
```
Comment (livecodepanos) [inline engine/src/srvwindows.cpp:295]: Shouldn't that be: 
```
return MCStringFormat(r_string, "NT %d.%d", MCOSVersionGetMinor(MCmajorosversion), MCOSVersionGetBugfix(MCmajorosversion));
``` 
?

File: engine/src/dskw32.cpp [hunk 5]
```diff
@@ -1617,7 +1617,7 @@ struct MCWindowsDesktop: public MCSystemInterface, public MCWindowsSystemService
     
 	virtual bool GetVersion(MCStringRef& r_string)
     {
-        return MCStringFormat(r_string, "NT %d.%d", (MCmajorosversion >> 8) & 0xFF, MCmajorosversion & 0xFF);
+        return MCStringFormat(r_string, "NT %d.%d", MCOSVersionGetMajor(MCmajorosversion), MCOSVersionGetMinor(MCmajorosversion));
     }
```
Comment (livecodepanos) [inline engine/src/dskw32.cpp:1621]: Shouldn't that be:

`return MCStringFormat(r_string, "NT %d.%d", MCOSVersionGetMinor(MCmajorosversion), MCOSVersionGetBugfix(MCmajorosversion));`

**Comment by runrevmark** [comment]
@livecodepanos The 0xAABB values in windows are AA.BB.00 - i.e. major, minor - not major.minor.maintenance... So 0x0501 => 5.1.0...

File: engine/src/dskw32.cpp (hunk 1 above)
Comment (livecodepanos) [reply engine/src/dskw32.cpp:368]: Oh I see - so in this case ignore all my other comments, since they are about the same issue :)

Comment (livecodeian) [reply engine/src/dskw32.cpp:368]: @livecodepanos the version format used on windows was 0xMMmm (MM = major, mm = minor) with no patch / bugfix version number, so 0x0501 => major = 5, minor = 1

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 2dd00f9

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 2dd00f9982940b9049eeb4d9c6ecb5be7f72f5a5

**Comment by livecode-vulcan** [comment]
😎 test success 2dd00f9982940b9049eeb4d9c6ecb5be7f72f5a5

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/926)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/334)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/333)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/332)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/333)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2451)
* try-community-universal-ios-iphoneos14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.1/builds/29)
* try-community-universal-ios-iphoneos13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/104)
* try-community-universal-ios-iphonesimulator14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.1/builds/27)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/100)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2478)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2467)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2465)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2502)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2438)

**Pull request merged**